CodeEdit/Features/SourceControl/Client/GitConfigClient.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  GitConfigClient.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 10/31/24.
+//
+
+import Foundation
+
+protocol GitConfigRepresentable {
+    init?(configValue: String)
+    var asConfigValue: String { get }
+}
+
+extension Bool: GitConfigRepresentable {
+    init?(configValue: String) {
+        switch configValue.lowercased() {
+        case "true": self = true
+        case "false": self = false
+        default: return nil
+        }
+    }
+
+    var asConfigValue: String {
+        self ? "true" : "false"
+    }
+}
+
+extension String: GitConfigRepresentable {
+    init?(configValue: String) {
+        self = configValue
+    }
+
+    var asConfigValue: String {
+        "\"\(self)\""
```

Comment:
A raw string is easier to read in this case:
```swift
#"\(self)"#
```

Reply:
This did not work. For some reason when a space existed it would add a separate entry.